control_panel_api/authentication.py

```diff
@@ -60,15 +60,38 @@ def get_or_create_user(decoded_payload):
     try:
         user = User.objects.get(pk=decoded_payload.get('sub'))
     except User.DoesNotExist:
-        user = User.objects.create(
-            pk=decoded_payload.get('sub'),
-            username=decoded_payload.get(settings.OIDC_FIELD_USERNAME),
-            email=decoded_payload.get(settings.OIDC_FIELD_EMAIL),
-            name=decoded_payload.get(settings.OIDC_FIELD_NAME),
-        )
-        user.save()
+        pass
```

Comment:
Are you sure about this logic? Maybe I'm missing something but the indentation may be off.

For example, we ignore the case when the user doesn't exist (it's new) because of the `pass`. But then outside the `exept` block we log `User "{}" is new [...]` which may not be the case.

Then we call `User.objects.create()` regardless of whethere the user was already in the DB or not. I don't remember how Django will handle that, it may raise an exception which we don't handle. Even if it doesn't we're just ignoring the result of `User.obects.get()`.

Reply:
My fault - I forgot to add a return when the user does exist. That changes the logic drastically!

Reply:
My fault - I forgot to add a return when the user does exist. That changes the logic drastically!